### PR TITLE
Add `docker build` step to the CI

### PR DESCRIPTION
**Pull request by neysofu**

### CODE REVIEW

**Comment by neysofu** [comment]
Fixes https://github.com/edgeandnode/graphix/issues/39.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,8 +1,16 @@
-name: CI
+name: CI pipeline
+
 on:
   push:
-    branches: main
+    branches: [master]
```
Comment (leoyvens) [inline .github/workflows/ci.yml:5]: Our main branch is called `main`.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
   pull_request:
+    types: [opened, synchronize, reopened]
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+  THEGRAPH_STORE_POSTGRES_DIESEL_URL: "postgresql://postgres:postgres@localhost:5432/graph_node_test"
```
Comment (leoyvens) [inline .github/workflows/ci.yml:12]: This env var is not relevant.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (neysofu) [reply .github/workflows/ci.yml:5]: Ops, I left that in from some copy & paste. Thank you for catching it!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (neysofu) [reply .github/workflows/ci.yml:12]: Same as above. Will remove it :+1:

**Pull request merged**